### PR TITLE
Label the eventing-publisher-proxy deployment with the current backendType

### PR DESCRIPTION
**Pull request by VladislavPaskar**
**Description**

To check if the event publisher proxy is ready, we need check it's pods. For that currently we fetch all the possible vent publisher proxy pods and check these pods by comparing the env vars.  This approach was introduced here https://github.com/kyma-project/kyma/pull/12312

Easier would be to label the publisher with its backend type, so they it be fetched and distinguished by this label.

Changes proposed in this pull request:

- label the eventing publisher proxy deployment with it's backendType 
- refactor the EPP deployment constructor using the functional options pattern

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/12606

### CODE REVIEW

**Comment by VladislavPaskar** [comment]
this PR requires change in reconciler, which cannot be performed before final kyma 2.x release, so it's blocked.

File: components/eventing-controller/pkg/deployment/publisher_deployment.go [hunk 1]
```diff
@@ -22,10 +23,11 @@ const (
...
 	bebNamespacePrefix       = "/"
-	instanceLabelKey         = "app.kubernetes.io/instance"
-	instanceLabelValue       = "eventing"
-	dashboardLabelKey        = "kyma-project.io/dashboard"
-	dashboardLabelValue      = "eventing"
+	InstanceLabelKey         = "app.kubernetes.io/instance"
+	InstanceLabelValue       = "eventing"
+	DashboardLabelKey        = "kyma-project.io/dashboard"
+	DashboardLabelValue      = "eventing"
+	BackendTypeLabelKey      = "backendType"
```
Comment (nachtmaar) [inline components/eventing-controller/pkg/deployment/publisher_deployment.go:30]: Kubernetes uses the prefix `app.kubernetes.io` ([source](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)).
|  Labels without a prefix are private to users.

Moreover, Kyma already comes with the `kyma-project.io` prefix. I suggest to either use this one or go one step further: `eventing.kyma-project.io`. What is your opinion on this ?

Comment (nachtmaar) [reply components/eventing-controller/pkg/deployment/publisher_deployment.go:30]: Or we use `kyma-project.io/eventingBackend`. backendType is very generic and could mean a lot and be used by other pods in Kyma as well in the future 🤔

Comment (VladislavPaskar) [reply components/eventing-controller/pkg/deployment/publisher_deployment.go]: I renamed the label name to `kyma-project.io/eventingBackend` as I think it's more suitable

**Review by nachtmaar — APPROVED** [review]

**Review by nachtmaar — APPROVED** [review]

**Review by nachtmaar — APPROVED** [review]

**Pull request merged**